assets/javascripts/discourse/components/chat-selection-manager.js

```diff
@@ -0,0 +1,117 @@
+import Component from "@ember/component";
+import discourseComputed from "discourse-common/utils/decorators";
+import { clipboardCopyAsync } from "discourse/lib/utilities";
+import { getOwner } from "discourse-common/lib/get-owner";
+import { action } from "@ember/object";
+import { ajax } from "discourse/lib/ajax";
+import { isTesting } from "discourse-common/config/environment";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { schedule } from "@ember/runloop";
+import { inject as service } from "@ember/service";
+import getURL from "discourse-common/lib/get-url";
+
+export default Component.extend({
+  tagName: "",
+  router: service(),
+  chatChannel: null,
+  selectedMessageIds: null,
+  showChatQuoteSuccess: false,
+  cancelSelecting: null,
+
+  @discourseComputed("selectedMessageIds")
+  anyMessagesSelected(selectedMessageIds) {
+    return selectedMessageIds.length > 0;
+  },
+
+  @action
+  async quoteMessages() {
```

Comment:
The general complexity of this action is very high, maybe we could split a few concerns

Reply:
Sure I could probably split it up into more functions, one for being isolated, one for mobile etc.